src/org/zaproxy/zap/extension/api/RubyAPIGenerator.java

```diff
@@ -0,0 +1,227 @@
+		for (ApiImplementor imp : implementors) {
+			this.generateRubyComponent(imp);
+		}
+	}
+
+	private void generateRubyComponent(ApiImplementor imp) throws IOException {
+		File f = new File(this.dir, createFileName(imp.getPrefix()));
+		System.out.println("Generating " + f.getAbsolutePath());
+		try (FileWriter out = new FileWriter(f)) {
+			out.write(HEADER);
+			out.write("# *** This file was automatically generated. ***\n\n");
+			out.write("module ZAP\n");
+			out.write("  class " + toTitleCase(safeName(imp.getPrefix())) + "\n\n");
+			out.write("    # @param zap [ZAP::Client]\n");
+			out.write("    def initialize(zap)\n");
+			out.write("      @zap = zap\n");
+			out.write("    end\n");
+			out.write("\n");
+
+			for (ApiElement view : imp.getApiViews()) {
+				this.generateRubyElement(view, imp.getPrefix(), "view", out);
+			}
+			for (ApiElement action : imp.getApiActions()) {
+				this.generateRubyElement(action, imp.getPrefix(), "action", out);
+			}
+			for (ApiElement other : imp.getApiOthers()) {
+				this.generateRubyElement(other, imp.getPrefix(), "other", out);
+			}
+			out.write("  end\n");
+			out.write("end");
+			out.write("\n");
+		}
+	}
+
+	private void generateRubyElement(ApiElement element, String component, String type, Writer out) throws IOException {
+
+		boolean typeOther = type.equals("other");
+		boolean hasParams = (element.getMandatoryParamNames() != null && element.getMandatoryParamNames().size() > 0)
+				|| (element.getOptionalParamNames() != null && element.getOptionalParamNames().size() > 0);
+
+		// Add description if defined
+		String descTag = element.getDescriptionTag();
+		if (descTag == null) {
+			// This is the default, but it can be overridden by the getDescriptionTag method if required
+			descTag = component + ".api." + type + "." + element.getName();
+		}
+		try {
+			String desc = msgs.getString(descTag);
+			out.write("    # " + desc + "\n");
+			if (optional) {
+				out.write("    #\n");
+				out.write("    # " + OPTIONAL_MASSAGE + "\n");
+			}
+		} catch (Exception e) {
+			// Might not be set, so just print out the ones that are missing
+			System.out.println("No i18n for: " + descTag);
+			if (optional) {
+				out.write("    # " + OPTIONAL_MASSAGE + "\n");
+			}
+		}
+
+		// Method declaration
+		out.write("    def " + createMethodName(element.getName()));
+		out.write("(");
+		
+		// Iterate through and write out method arguments
+		ArrayList<String> args = new ArrayList<String>();
+		if (element.getMandatoryParamNames() != null) {
+			for (String param : element.getMandatoryParamNames()) {
+				args.add(param.toLowerCase());
+			}
+		}
+		if (element.getOptionalParamNames() != null) {
+			for (String param : element.getOptionalParamNames()) {
+				args.add(param.toLowerCase() + " = ''");
+			}
+		}
+		out.write(StringUtils.join(args, ", "));
+		out.write(")\n");
+
+		String method = "request";
+		if (typeOther) { method += "_other"; }
```

Comment:
As a matter of consistency, the if statement should be indented as the other if statements (i.e. statements of if block and the closing brace should be in a new line).